dataclients/kubernetes/kubernetestest/api_test.go

```diff
@@ -235,117 +256,108 @@ func TestTestAPI(t *testing.T) {
 		check(t, sec, 1, "Secret")
 	})
 
 	t.Run("without namespace", func(t *testing.T) {
 		var s map[string]interface{}
-		if err := get(kubernetes.ServicesClusterURI, &s); err != nil {
-			t.Fatal(err)
-		}
+		get(t, kubernetes.ServicesClusterURI, &s)
 		check(t, s, 3, "Service")
 
 		var i map[string]interface{}
-		if err := get(kubernetes.IngressesV1ClusterURI, &i); err != nil {
-			t.Fatal(err)
-		}
+		get(t, kubernetes.IngressesV1ClusterURI, &i)
 		check(t, i, 1, "Ingress")
 
 		var r map[string]interface{}
-		if err := get(kubernetes.RouteGroupsClusterURI, &r); err != nil {
-			t.Fatal(err)
-		}
+		get(t, kubernetes.RouteGroupsClusterURI, &r)
 		check(t, r, 2, "RouteGroup")
 
 		var e map[string]interface{}
-		if err := get(kubernetes.EndpointsClusterURI, &e); err != nil {
-			t.Fatal(err)
-		}
+		get(t, kubernetes.EndpointsClusterURI, &e)
 		check(t, e, 3, "Endpoints")
 
 		var eps map[string]interface{}
-		if err := get(kubernetes.EndpointSlicesClusterURI, &eps); err != nil {
-			t.Fatal(err)
-		}
+		get(t, kubernetes.EndpointSlicesClusterURI, &eps)
 		check(t, eps, 3, "EndpointSlice")
 
 		var sec map[string]interface{}
-		if err := get(kubernetes.SecretsClusterURI, &sec); err != nil {
-			t.Fatal(err)
-		}
+		get(t, kubernetes.SecretsClusterURI, &sec)
 		check(t, sec, 1, "Secret")
 	})
 
 	t.Run("kind: List", func(t *testing.T) {
 		const namespace = "baz"
 
 		var s map[string]interface{}
-		if err := get(fmt.Sprintf(kubernetes.ServicesNamespaceFmt, namespace), &s); err != nil {
-			t.Fatal(err)
-		}
+		get(t, fmt.Sprintf(kubernetes.ServicesNamespaceFmt, namespace), &s)
 		check(t, s, 1, "Service")
 
 		var i map[string]interface{}
-		if err := get(fmt.Sprintf(kubernetes.IngressesV1NamespaceFmt, namespace), &i); err != nil {
-			t.Fatal(err)
-		}
+		get(t, fmt.Sprintf(kubernetes.IngressesV1NamespaceFmt, namespace), &i)
 		check(t, i, 0, "Ingress")
 
 		var r map[string]interface{}
-		if err := get(fmt.Sprintf(kubernetes.RouteGroupsNamespaceFmt, namespace), &r); err != nil {
-			t.Fatal(err)
-		}
+		get(t, fmt.Sprintf(kubernetes.RouteGroupsNamespaceFmt, namespace), &r)
 		check(t, r, 1, "RouteGroup")
 
 		var e map[string]interface{}
-		if err := get(fmt.Sprintf(kubernetes.EndpointsNamespaceFmt, namespace), &e); err != nil {
-			t.Fatal(err)
-		}
+		get(t, fmt.Sprintf(kubernetes.EndpointsNamespaceFmt, namespace), &e)
 		check(t, e, 1, "Endpoints")
 
 		var eps map[string]interface{}
-		if err := get(fmt.Sprintf(kubernetes.EndpointSlicesNamespaceFmt, namespace), &eps); err != nil {
-			t.Fatal(err)
-		}
+		get(t, fmt.Sprintf(kubernetes.EndpointSlicesNamespaceFmt, namespace), &eps)
 		check(t, eps, 1, "EndpointSlice")
 
 		var sec map[string]interface{}
-		if err := get(fmt.Sprintf(kubernetes.SecretsNamespaceFmt, namespace), &sec); err != nil {
-			t.Fatal(err)
-		}
+		get(t, fmt.Sprintf(kubernetes.SecretsNamespaceFmt, namespace), &sec)
 		check(t, sec, 0, "Secret")
 	})
+
+	t.Run("resource by name", func(t *testing.T) {
```

Comment:
Should we add "resource with this name does not exist in current namespace, but exists in another"?

Reply:
Why? The result will be the same as "resource name does not exist" below

Reply:
I expected to have some probability of some bug related to it in k8s API, but maybe I am too paranoid.